### PR TITLE
Courses table preview action

### PR DESCRIPTION
**Pull request by nunibaranes**
- Created components: `Modal` , `PreviewModal` and `PreviewCoursesButton`.

- `Modal` is a generic component, creating 'root-modal' DOM element next to app's 'root' element,  using ReactDOM.createPortal. 
- `PreviewModal` is rendering header and iframe to preview the selected items, (currently I'm rendering static url - the SDK is not ready )
- `PreviewCoursesButton` clicking on 'preview' button, manage local state to manage the modal and child behavior.
- Adding animation.
- commented the `PreviewCoursesButton` component. to prevent rendering the feature before is ready.

### CODE REVIEW

File: src/hooks/useWindowResize.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useState } from 'react';
+
+export const useWindowResize = (): {
+  windowWidth: number;
+  windowHeight: number;
```
Comment (dr15) [inline src/hooks/useWindowResize.tsx:5]: Consider using `useEventListener` for the implementation of `useWindowResize`. Also consider using throttle on the event callback.

File: src/components/Screen/CoursesScreen/PreviewCoursesButton.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import classes from './style.module.scss';
+
+function PreviewCoursesButton({
+  selectedRows,
+  dispatch,
+}: {
+  selectedRows: Array<UICourse>;
+  dispatch: Dispatch<any>;
+}): ReactElement {
+  const [showModal, setModal] = useState(false);
```
Comment (dr15) [inline src/components/Screen/CoursesScreen/PreviewCoursesButton.tsx:18]: Please change `setModal` to `setShowModal`

File: src/components/Screen/CoursesScreen/PreviewCoursesButton.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  const [showPreviewCoursesModal, setShowPreviewCoursesModal] = useState(false);
+  const [isFullScreen, setIsFullScreen] = useState(false);
+
+  const onPreviewCourses = () => {
```
Comment (dr15) [inline src/components/Screen/CoursesScreen/PreviewCoursesButton.tsx:22]: Please add `useCallback()` to any callback you pass to a child component

File: src/components/Layout/Modal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  onClose,
+}: {
+  children: ReactNode;
+  isOpen: boolean;
+  closable?: boolean;
+  onClose?: () => void;
+}): ReactElement {
+  const [containerEl] = useState(document.getElementById('modal-root'));
+
+  const setGlobals = useCallback(() => {
```
Comment (dr15) [inline src/components/Layout/Modal/index.tsx:23]: When creating a function that is used within the same component there is no need to ass `useCallback()`, it is only necessary when passin it to a child component

**Review by dr15 — CHANGES_REQUESTED** [review]
Please look at the comments I left

File: src/components/Layout/Modal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  useEffect(() => {
+    containerEl?.classList.add(classes['wm-modal']);
+  }, []);
+
+  useEffect(() => {
+    if (isOpen) {
+      containerEl?.classList.add(classes['is-open']);
+    } else {
+      containerEl?.classList.remove(classes['is-open']);
+    }
```
Comment (RonnyO) [inline src/components/Layout/Modal/index.tsx:33]: Why not `containerEl?.classList.toggle(classes['is-open'], isOpen);`? See example [here](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#Examples).

File: src/hooks/useWindowResize.tsx (hunk 1 above)
Comment (dvira) [reply src/hooks/useWindowResize.tsx]: We added a library (a prewritten hook) that does window size calculation with debounce. 
The `useEventListener` does not accept a condition for adding the event listener, so we kept this implementation for now, until we expand the hook's api.

File: src/components/Screen/CoursesScreen/PreviewCoursesButton.tsx (hunk 2 above)
Comment (dvira) [reply src/components/Screen/CoursesScreen/PreviewCoursesButton.tsx]: Done

File: src/components/Screen/CoursesScreen/PreviewCoursesButton.tsx (hunk 3 above)
Comment (dvira) [reply src/components/Screen/CoursesScreen/PreviewCoursesButton.tsx]: Right, done.

**Review by dr15 — APPROVED** [review]
🔥

**Pull request merged**